testing/trino-testing/src/main/java/io/trino/testing/containers/HiveHadoop.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.testing.containers;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.net.HostAndPort;
+import org.testcontainers.containers.Network;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+public class HiveHadoop
+        extends BaseTestContainer
+{
+    public static final String DEFAULT_IMAGE = "ghcr.io/trinodb/testing/hdp2.6-hive:41";
+    public static final String HOST_NAME = "hadoop-master";
+
+    public static final int PROXY_PORT = 1180;
+    public static final int HIVE_SERVER_PORT = 10000;
+    public static final int HIVE_METASTORE_PORT = 9083;
+
+    public static Builder builder()
+    {
+        return new Builder();
+    }
+
+    private HiveHadoop(
+            String image,
+            String hostName,
+            Set<Integer> ports,
+            Map<String, String> filesToMount,
+            Map<String, String> envVars,
+            Optional<Network> network,
+            int startupRetryLimit)
+    {
+        super(
+                image,
+                hostName,
+                ports,
+                filesToMount,
+                envVars,
+                network,
+                startupRetryLimit);
+    }
+
+    @Override
+    protected void setupContainer()
+    {
+        super.setupContainer();
+        String runCmd = "/usr/local/hadoop-run.sh";
+        copyFileToContainer("io/trino/testing/containers/hadoop-run.sh", runCmd);
+        withRunCommand(
+                ImmutableList.of(
+                        "/bin/bash",
+                        runCmd));
+    }
+
+    public HostAndPort getMappedHdfsSocksProxy()
```

Comment:
this is not Hdfs specific.
I would call method `getSocksProxyEndpoint`

Reply:
Done